openapi_python_client/parser/properties/model_property.py

```diff
@@ -14,6 +14,17 @@
 from .schemas import Class, ReferencePath, Schemas, parse_reference_path
 
 
+@define
+class ModelDetails:
+    """Container for basic attributes of a model schema that can be computed separately"""
+
+    required_properties: list[Property] | None = None
+    optional_properties: list[Property] | None = None
+    additional_properties: Property | None = None
+    relative_imports: set[str] = field(factory=set)
+    lazy_imports: set[str] = field(factory=set)
```

Comment:
This is similar to the `_PropertyData` class that previously existed just to be a return type for `_process_properties`. Since they were so close already, and `ModelProperty` also had all of these properties, I figured it made sense to encapsulate them and have `ModelProperty` just own an instance of this class. I added accessor methods so that `model.required_properties` still works as an alias for `model.details.required_properties`.